### PR TITLE
Reduce boilerplate in the sketch

### LINKED ISSUE #5: Reduce boilerplate in sketch code

### PR DESCRIPTION
**Pull request by sandeepmistry**
To start the discussion to resolve #5 and #6. This will also allow us to have unified examples for all platforms.

Later on, we could also use this class to make the Azure SDK API's more Arduino style instead of C.

@stefangordon @obsoleted please review and provide feedback. I've tested on a MKR1000, and made sure the new example compiles on other supported platforms.

### CODE REVIEW

File: src/AzureIoTHub.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+#error "Unsupported platform"
+#endif
+
+#include "util/NTPClient.h"
+
+#include "AzureIoTHub.h"
+
+AzureIoTHubClass AzureIoTHub;
+
+#if defined(ARDUINO_SAMD_FEATHER_M0)
```
Comment (sandeepmistry) [inline src/AzureIoTHub.cpp]: @ladyada there is potential to remove use of ```Adafruit_WINC1500``` library in this library and just use ```WiFi101``` if the variant for the "Adafruit Feather M0 WiFi" board defines following WINC1500 pin values like the MKR1000: https://github.com/arduino/ArduinoCore-samd/blob/master/variants/mkr1000/variant.h#L137-L143

**Comment by ladyada** [comment]
ok - i'll try it!
can we also update all the examples to not have a 10 second wait on wifi connect? its often 2 seconds, and it makes the WINC seems slower than it really is :)

https://github.com/adafruit/Adafruit_WINC1500/commit/7748c72cb083fa3fba6706afaed39ebe82fd77f5#diff-1425a7d9293c2f199af50199fa6cc69fL36

**Comment by sandeepmistry** [comment]
> can we also update all the examples to not have a 10 second wait on wifi connect? its often 2 seconds, and it makes the WINC seems slower than it really is :)

@ladyada please create an issue in the WiFi101 library to discuss this. From my testing the delay seems unnecessary for unsuccessful connection. I've been using:
```arduino
  while (WiFi.begin(ssid, pass) != WL_CONNECTED) {
    // unsuccessful, retry in 4 seconds
    Serial.print("failed ... ");
    delay(4000);
    Serial.print("retrying ... ");
  }
```

**Comment by sandeepmistry** [comment]
After discussing with @cmaglie we should not have WiFi configuration code inside the library. I'm closing this and will propose something else.

**Comment by ladyada** [comment]
@sandeepmistry ok will make new issue for delay thingy, i definitely noticed delay makes a difference and it drove me a little nuts to wait :dancer: 

 as for the WiFi101 thingy, whatever you like. i strongly prefer passing in the pin names as args (and you can default them if you like) but just lemme know and ill update/delete fork/whatever :+1:

**Comment by sandeepmistry** [comment]
> as for the WiFi101 thingy, whatever you like. i strongly prefer passing in the pin names as args (and you can default them if you like) but just lemme know and ill update/delete fork/whatever 

@ladyada let's get a discussion going in the WiFi101 repo for this as well

**Pull request closed without merging**